### ISSUE TITLE
[readability/casting] false positive with typedef

### ISSUE THREAD

**Issue by targos** [cpplint/cpplint#409]
In Node.js, we have this code that triggers the `readability/casting error`:

```c++
typedef int32_t(NAPI_CDECL* node_api_addon_get_api_version_func)(void);
```

https://github.com/nodejs/node/blob/6706b22e350490d3e8e697377b5f388df9305aa0/src/node_api_types.h#L8

> src/node_api_types.h:8:  Using deprecated casting style.  Use static_cast<int32_t>(...) instead  [readability/casting] [4]

**Comment by aaronliu0130** [comment]
Also fails:
```cpp
using node_api_addon_get_api_version_func = int32_t(NAPI_CDECL *)(void);
```
